src/core/utils/channel_manager.cpp

```diff
@@ -85,11 +93,27 @@ void ChannelManager::RequestChannelChange(uint8_t aChannel)
     mTimer.Start(1 + Random::NonCrypto::GetUint32InRange(0, kRequestStartJitterInterval));
 
     Get<Notifier>().Signal(kEventChannelManagerNewChannelChanged);
+#else // OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
```

Comment:
I can think of an issue: An FTD can also work as an SSED. In this case, the auto channel selection should choose CSL channel instead of PAN channel. We need to handle this case. So I think it cannot be totally determined in compile-time.

Reply:
Thanks for your comment. I added some code to handle this case.